src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -0,0 +1,120 @@
+#include "software/backend/simulation/physics/physics_robot.h"
+
+#include "shared/constants.h"
+#include "software/backend/simulation/physics/box2d_util.h"
+
+PhysicsRobot::PhysicsRobot(std::shared_ptr<b2World> world, const Robot& robot)
+    : robot_id(robot.id())
+{
+    // All the BodyDef must be defined before the body is created.
```

Comment:
what's a `BodyDef`?

Reply:
Also I'm a bit confused about the order stuff is happening here. How do we know this is "before the body is created"?

Reply:
Changed this comment as it wasn't appropriate here. Clarified with a new comment.

tl;dr a `BodyDef` is a struct that defines properties of a `b2Body` in the physics engine.